### PR TITLE
Updated configuration comments and documentation

### PR DESCRIPTION
**Pull request by stevenengler**
Update the documentation for the new YAML config format.

### CODE REVIEW

**Review by robgjansen — APPROVED** [review]
Overall, this looks reasonable.

I requested some terminology changes so that we don't refer to processes as 'simulated' processes, since they will in fact be real Linux processes. I ignored some instances in the `Shadow-Config.md` file because I assumed that file would be updated after you updated `configuration.rs`.

Do we want to document the process for generating a new version of `Shadow-Config.md` file after making changes to the docs in `configuration.rs` for the enxt time we need to generate it?

File: docs/3.1-Shadow-Config.md [hunk 1]
```diff
@@ -1,83 +1,461 @@
...
+Logs all network input and output for this host in PCAP format (for viewing in e.g. wireshark).
+
+#### `host_defaults.type_hint`
+
+Default: null  
+Type: String OR null
+
+Node type hint for Shadow's name and routing system.
+
+This hint will be used to assign the host to a node based on the types of nodes in the network topology.
```
[the remark this answers is not in the document]
Comment (stevenengler) [reply docs/3.1-Shadow-Config.md:390]: I've made a separate issue for this: #1326

File: docs/3.1-Shadow-Config.md [hunk 2]
```diff
@@ -1,83 +1,461 @@
...
+*Required*  
+Type: "plain" OR "graphml" OR "path"
 
-_cpufrequency_ is the speed of this _host's_ virtual CPU in kilohertz. Along with the CPU processing requirements of the plug-in process, this determines how often events for this _host_ are delayed during simulation.
+The topology can be specified as a built-in "plain" topology with a single node, an inline GraphML string, or a path to an external GraphML file.
 
-_logpcap_ is a case insensitive boolean string (e.g. "true") that specifies that Shadow should log all network input and output for this _host_ in PCAP format (for viewing in e.g. wireshark). _pcapdir_ is the directory to which the logs should be saved for this _host_.
+The topology, a collection of nodes and edges that represent a network topology, should hold an undirected, complete, connected graph with certain attributes specified on the nodes and edges. For more information on how to structure this data, see the [Topology Format](3.2-Network-Config.md).
```
[the remark this answers is not in the document]
Comment (stevenengler) [reply docs/3.1-Shadow-Config.md:145]: I removed the "complete" here.

File: src/main/core/support/configuration.rs [hunk 3]
```diff
@@ -437,19 +439,24 @@ impl Default for HostDefaultOptions {
...
 
+    /// Number of processes to start
     #[serde(default)]
     quantity: Quantity,
 
+    /// The simulated time at which to start the simulated process
     #[serde(default)]
     start_time: units::Time<units::TimePrefixUpper>,
 
+    /// The simulated time at which this simulated process is sent a SIGKILL signal
```
[the remark this answers is not in the document]
Comment (stevenengler) [reply src/main/core/support/configuration.rs]: Fixed

File: docs/3.1-Shadow-Config.md [hunk 4]
```diff
@@ -1,83 +1,461 @@
...
+Node type hint for Shadow's name and routing system.
+
+This hint will be used to assign the host to a node based on the types of nodes in the network topology.
+
+#### `hosts`
+
+*Required*
+Type: Object
+
+The simulated hosts which run the simulated processes. Each field corresponds to a host configuration, with the field name being used as the network hostname.
```
[the remark this answers is not in the document]
Comment (stevenengler) [reply docs/3.1-Shadow-Config.md]: Fixed

**Pull request merged**